app/grandchallenge/components/views.py

```diff
@@ -270,6 +276,29 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class CIVSetDetail(
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    DetailView,
+):
+    model = None
+    permission_required = None
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+    template_name = "components/civ_set_detail.html"
+
+    @cached_property
+    def base_object(self):
+        return NotImplementedError
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+
+        context.update(dict(base_object=self.base_object))

```

Comment:
You don't need to add this to the context, do you? Since this is a detail view, there will always be an object for which you can access the base_object in the template by simply doing `object.base_object`.

Reply:
This is in-line with the other CIV template styles.

Reply:
In those other views, it is needed though, because you don't have a single object to get the base_object from (yet) (i.e. on the listview and the formview). The delete view also does this unnecessarily, I just saw, that could be refactored (we also have a duplicate template there for some reason, but that can be fixed in a seperate PR).

Reply:
Updated!